mne/preprocessing/hfc.py

```diff
@@ -0,0 +1,135 @@
+# Authors: George O'Neill <g.o'neill@ucl.ac.uk>
+#
+# License: BSD-3-Clause
+
+import numpy as np
+from .maxwell import _prep_mf_coils, _sss_basis
+from ..forward import _prep_meg_channels
+from ..io.pick import pick_types
+from ..io.proj import Projection
+from ..utils import verbose
+
+
+@verbose
+def compute_proj_hfc(info, order=1, picks=None, exclude_bads=True,
+                     accuracy='accurate', verbose=None):
+    """Generate projectors to perform homogeneous/harmonic correction to data.
+
+    Remove evironmental fields from magentometer data by assuming it is
+    explained as a homogeneous :footcite:`TierneyEtAl2021` or harmonic field
+    :footcite:`TierneyEtAl2022`. Useful for arrays of OPMs.
+
+    Parameters
+    ----------
+    info : instance of Info
+        Info from an instance of Raw (preferable).
+    order : int
+        The order of the spherical harmonic basis set to use. Set to 1 to use
+        only the homogeneous field component (default), 2 to add gradients, 3
+        to add quadrature terms etc.
+    picks : int | array-like
+        List of channels indices to compute projeciton on. Default behaviour
+        is to select meg (non-reference) with known positions.
+    exclude_bads : bool
+        Do not include bad channels in the projection, (default: True).
+    accuracy : str
+        Can be ``"point"``, ``"normal"`` or ``"accurate"`` (default), defines
+        which level of coil definition accuracy is used to generate model.
+    %(verbose)s
+
+    Returns
+    -------
+    projs: list
+        List of projection vectors.
+
+    Notes
+    -----
+    To apply the projectors to a dataset, use
+    ``[DATA].add_proj(projs).apply_proj()``.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+    if picks is None:
+        if exclude_bads is True:
+            picks = pick_types(info, meg=True)
+        else:
+            picks = pick_types(info, meg=True, exclude=None)
+    picks = _filter_channels_with_positions(info, picks)
+    basis, channels = _generate_basis_set(info, picks, order, accuracy)
+    labels = _label_basis(order)
+    projs = []
+    for ii in range(len(labels)):
```

Comment:
Let's save ourselves at least one indexing operation
```suggestion
    for ii, label in enumerate(labels):
```